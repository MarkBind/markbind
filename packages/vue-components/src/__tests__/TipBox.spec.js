import { mount } from '@vue/test-utils';
import TipBox from '../TipBox.vue';

describe('TipBox', () => {
  test('of default type with content renders correctly ', () => {
    const wrapper = mount(TipBox, {
      slots: {
        default: 'This is the default box',
      },
    });
    // TODO: Consider whether to make the test more robust and specific
    // for example: expect(wrapper.classes()).toContain('alert-default');
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of info type renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'info',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of warning type renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'warning',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of success type renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'success',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of important type renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'important',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of wrong type renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'wrong',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of tip type renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'tip',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of definition type renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'definition',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('with dismissible option renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        dismissible: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('with header renders correctly', () => {
    const wrapper = mount(TipBox, {
      slots: {
        _header: 'A header',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of info type light style renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'info',
        light: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of warning type light style renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'warning',
        light: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of success type light style renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'success',
        light: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of important type light style renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'important',
        light: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of wrong type light style renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'wrong',
        light: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of tip type light style renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'tip',
        light: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of definition type light style renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'definition',
        light: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('of seamless style renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        seamless: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('having custom background color renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        backgroundColor: 'white',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('having custom border color renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        borderColor: 'grey',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('having custom border left color renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        borderLeftColor: 'blue',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('should have light style take priority over seamless', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        light: true,
        seamless: true,
      },
    });
    const boxLightStyle = mount(TipBox, {
      propsData: {
        light: true,
      },
    });
    expect(wrapper).toEqual(boxLightStyle);
  });

  test('with no-background option renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        noBackground: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('with no-background option does not affect background-color', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        noBackground: true,
        backgroundColor: 'yellow',
      },
    });
    const boxYellowBackground = mount(TipBox, {
      propsData: {
        backgroundColor: 'yellow',
      },
    });
    expect(wrapper).toEqual(boxYellowBackground);
  });

  test('with no-border option renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        noBorder: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('with no-border option does not affect border-color', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        noBorder: true,
        borderColor: 'grey',
      },
    });
    const boxGreyBorder = mount(TipBox, {
      propsData: {
        borderColor: 'grey',
      },
    });
    expect(wrapper).toEqual(boxGreyBorder);
  });

  test('with no-border option does not affect border-left-color', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        noBorder: true,
        borderLeftColor: 'grey',
      },
    });
    const boxGreyLeftBorder = mount(TipBox, {
      propsData: {
        borderLeftColor: 'grey',
      },
    });
    expect(wrapper).toEqual(boxGreyLeftBorder);
  });

  test('with no-icon option renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        type: 'success',
        noIcon: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('with icon renders correctly', () => {
    const wrapper = mount(TipBox, {
      slots: {
        icon: ':rocket:',
      },
      propsData: {
        type: 'info',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('with no-icon option does not affect icon set by icon', () => {
    const wrapper = mount(TipBox, {
      slots: {
        icon: ':rocket:',
      },
      propsData: {
        type: 'info',
        noIcon: true,
      },
    });
    const boxWithIcon = mount(TipBox, {
      slots: {
        icon: ':rocket:',
      },
      propsData: {
        type: 'info',
      },
    });
    expect(wrapper).toEqual(boxWithIcon);
  });

  test('with icon and icon-size renders correctly', () => {
    const wrapper = mount(TipBox, {
      slots: {
        icon: ':rocket:',
      },
      propsData: {
        type: 'info',
        iconSize: '2x',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('with custom color renders correctly', () => {
    const wrapper = mount(TipBox, {
      propsData: {
        color: 'pink',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('with custom icon color and header renders correctly', () => {
    const wrapper = mount(TipBox, {
      slots: {
        icon: ':fas-plus:',
        _header: 'A header',
      },
      propsData: {
        iconColor: 'red',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('with custom icon color and no header renders correctly', () => {
    const wrapper = mount(TipBox, {
      slots: {
        icon: ':fas-plus:',
      },
      propsData: {
        iconColor: 'red',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
