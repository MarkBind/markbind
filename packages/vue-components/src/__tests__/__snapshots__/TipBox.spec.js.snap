// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TipBox having custom background color renders correctly 1`] = `
<div
  class="alert box-container alert-default"
  style="background-color: white; border-color: white;"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <!---->
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox having custom border color renders correctly 1`] = `
<div
  class="alert box-container alert-default"
  style="border: 1px solid grey;"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <!---->
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox having custom border left color renders correctly 1`] = `
<div
  class="alert box-container alert-default"
  style="border-left: 5px solid blue;"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <!---->
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of default type with content renders correctly  1`] = `
<div
  class="alert box-container alert-default"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <!---->
     
    <!---->
     
    <div
      class="contents"
    >
      This is the default box
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of definition type light style renders correctly 1`] = `
<div
  class="alert box-container border-primary alert-border-left"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper  text-primary"
    >
      <i
        class="fas fa-atlas"
      />
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of definition type renders correctly 1`] = `
<div
  class="alert box-container alert-primary"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper"
    >
      <i
        class="fas fa-atlas"
      />
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of important type light style renders correctly 1`] = `
<div
  class="alert box-container border-danger alert-border-left"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper  text-danger"
    >
      <i
        class="fas fa-flag"
      />
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of important type renders correctly 1`] = `
<div
  class="alert box-container alert-danger"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper"
    >
      <i
        class="fas fa-flag"
      />
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of info type light style renders correctly 1`] = `
<div
  class="alert box-container border-info alert-border-left"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper  text-info"
    >
      <i
        class="fas fa-info"
      />
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of info type renders correctly 1`] = `
<div
  class="alert box-container alert-info"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper"
    >
      <i
        class="fas fa-info"
      />
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of seamless style renders correctly 1`] = `
<div
  class="alert box-container seamless"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <!---->
     
    <div
      aria-hidden="true"
      class="vertical-divider alert-default"
    />
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of success type light style renders correctly 1`] = `
<div
  class="alert box-container border-success alert-border-left"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper  text-success"
    >
      <i
        class="fas fa-check"
      />
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of success type renders correctly 1`] = `
<div
  class="alert box-container alert-success"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper"
    >
      <i
        class="fas fa-check"
      />
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of tip type light style renders correctly 1`] = `
<div
  class="alert box-container border-success alert-border-left"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper  text-success"
    >
      <i
        class="fas fa-lightbulb"
      />
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of tip type renders correctly 1`] = `
<div
  class="alert box-container alert-success"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper"
    >
      <i
        class="fas fa-lightbulb"
      />
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of warning type light style renders correctly 1`] = `
<div
  class="alert box-container border-warning alert-border-left"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper  text-warning"
    >
      <i
        class="fas fa-exclamation"
      />
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of warning type renders correctly 1`] = `
<div
  class="alert box-container alert-warning"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper"
    >
      <i
        class="fas fa-exclamation"
      />
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of wrong type light style renders correctly 1`] = `
<div
  class="alert box-container border-danger alert-border-left"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper  text-danger"
    >
      <i
        class="fas fa-times"
      />
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox of wrong type renders correctly 1`] = `
<div
  class="alert box-container alert-danger"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper"
    >
      <i
        class="fas fa-times"
      />
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox with custom color renders correctly 1`] = `
<div
  class="alert box-container alert-default"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <!---->
     
    <!---->
     
    <div
      class="contents"
      style="color: pink;"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox with dismissible option renders correctly 1`] = `
<div
  class="alert box-container alert-default"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper alert-dismissible"
  >
    <!---->
     
    <!---->
     
    <div
      class="contents"
    />
     
    <button
      aria-label="Close"
      class="close"
      data-dismiss="alert"
      type="button"
    >
      <span
        aria-hidden="true"
      >
        Ã—
      </span>
    </button>
  </div>
</div>
`;

exports[`TipBox with header renders correctly 1`] = `
<div
  class="alert box-container alert-default"
>
  <div
    class="box-header-wrapper"
  >
    <!---->
     
    <div
      class="box-header"
    >
      A header
    </div>
     
    <!---->
  </div>
   
  <!---->
   
  <div
    class="box-body-wrapper box-body-wrapper-with-heading"
  >
    <!---->
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox with icon and icon-size renders correctly 1`] = `
<div
  class="alert box-container alert-info"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper fa-2x"
    >
      :rocket:
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox with icon renders correctly 1`] = `
<div
  class="alert box-container alert-info"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <div
      class="icon-wrapper"
    >
      :rocket:
    </div>
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox with no-background option renders correctly 1`] = `
<div
  class="alert box-container alert-default no-background"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <!---->
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox with no-border option renders correctly 1`] = `
<div
  class="alert box-container alert-default no-border"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <!---->
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;

exports[`TipBox with no-icon option renders correctly 1`] = `
<div
  class="alert box-container alert-success"
>
  <!---->
   
  <!---->
   
  <div
    class="box-body-wrapper"
  >
    <!---->
     
    <!---->
     
    <div
      class="contents"
    />
     
    <!---->
  </div>
</div>
`;
